tensorflow_addons/layers/adaptive_pooling.py

```diff
@@ -15,12 +15,185 @@
 """Pooling layers with fixed size outputs"""
 
 import tensorflow as tf
+import tensorflow_addons.utils.keras_utils as conv_utils
+
 from typeguard import typechecked
-from typing import Union, List, Tuple
+from typing import Union, Callable, Iterable
+
+
+class AdaptivePooling1D(tf.keras.layers.Layer):
```

Comment:
My thought is to put `get_config` under `AdaptivePooling{1,2,3}D` so that we do not need to override it every time. What do you think? @Susmit-A

Reply:
Sounds good. I'll move it.